### DISCUSSION TITLE
Error - No subcommand group specified for interaction. While Having SubcommandGroup and Subcommand ( root )

### DISCUSSION THREAD

**Discussion by Batman4234** [discordjs/discord.js#7761] [Q&A] [answered]
### Which package is this bug report for?

discord.js

### Issue description

f we got code like the one i provided  and do ```const subcmdg = interaction.options.getSubcommandGroup()``` it gives the error ```TypeError [COMMAND_INTERACTION_OPTION_NO_SUB_COMMAND_GROUP]: No subcommand group specified for interaction.
    at CommandInteractionOptionResolver.getSubcommandGroup (/home/ibraim6/Documents/Programming/woodo-bot/node_modules/discord.js/src/structures/CommandInteractionOptionResolver.js:124:13)
    at Command.run (/home/ibraim6/Documents/Programming/woodo-bot/src/commands/blacklist/something.ts:23:45)
    at ExtendedClient.<anonymous> (/home/ibraim6/Documents/Programming/woodo-bot/src/events/interactionCreate.ts:13:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[ERROR] 21:51:26 TypeError [COMMAND_INTERACTION_OPTION_NO_SUB_COMMAND_GROUP]: No subcommand group specified for interaction.```

### Code sample

```typescript
options: [{
        name: 'random',
        description: 'random',
        type: 'SUB_COMMAND_GROUP',
        options: [{
            name: 'random',
            description: 'random',
            type: 'SUB_COMMAND'
        },
    
    ]
    },{
        name: 'randoms',
        description: 'randoms',
        type: 'SUB_COMMAND'
    }],
// in run
const subcmdg = interaction.options.getSubcommandGroup()
```

### Package version

13.5.0

### Node.js version

v16.14.0

### Operating system

Fedora 35

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

No Intents

**Comment by Jiralite** [comment]
This isn't a bug. Your bot is receiving the application command that is not a subcommand group, which is only one other command as seen from your reproducible code sample.

**Comment by imranbarbhuiya** [comment, marked as answer]
Subcommand group and Subcommand are default to `required = true`. So when a user uses your Subcommand but not subcommandgroup, it'll throw error. This was changed in main branch so in v14 for Subcommandgroup `required` is default to false. For now u can pass `required = false` like this to avoid the error

```js
interaction.options.getSubcommandGroup(false)
```


U can check this pr to know more
https://github.com/discordjs/discord.js/pull/7217

**Comment by Batman4234** [comment]
oh thanks imranbhuiya
